docs/contributing/999-rfcs/2023-10-01-workloads.md

```diff
@@ -0,0 +1,262 @@
+---
+title: "#4350 Containerized Workloads - Requirements"
+description: Requirements for containerized workloads in the Wing Cloud Library
+---
+
+# Containerized Workloads - Requirements Specification
+
+- **Author(s)**: @eladb
+- **Submission Date**: 2023-10-01
+- **Stage**: Proposal
+- **Stage Date**: 2023-10-01
+
+This specification lists the *requirements* we have identified so far that are related to using
+containers to implement long running workloads (aka "services" or "microservices"). 
+
+## Requirements
+
+Requirements are marked with a *Reqtag* so they will be easy to trace in subsequent design
+documents. Suggested priorities are marked inline as "(P1)", "(P2)" or "(P3)".
+
+### Containerized workloads (P1)
+
+> Reqtag: `c:workload`
+
+A "workload" describes a unit of scalable deployment. It's akin to a K8S's deployment + service + ingress, or ECS's task or fly.io's machine or ControlPlane's workload.
+Wing must support defining **containerized workloads** implemented through user-defined container
+images. Images may either be pulled from a container registry or built from a `Dockerfile` within
+the project directory.
+
+### Simulator implementation (P1)
+
+> Reqtag: `c:simulator`
+
+When using Docker images, the Wing simulator will need to take care of seamlessly building and/or
+pulling the container image and maintaining it's lifecycle and cleanup via a local docker
+installation.
+
+If docker is not installed, an error should provide useful instructions on how to install it.
```

Comment:
Since we plan to inform the user with useful instructions on how to install docker, should we also provide them with useful instructions on how to ensure they have permissions to push/pull from container registries (in the event of a permissions issue)? 

I.E. if they are using ECR/GCR/etc.., we (Wing) are able to configure permissions to ECS/K8/etc.. but for local testing that is on them to have a configured credentials (I assume the same way its on them to have docker installed)